Allow us to use system googletest please.

You have hard coded to clone googletest. In debian we cannot make network calls.

CMake Error at /usr/share/cmake-3.13/Modules/ExternalProject.cmake:2329 (message):
  error: could not find git for clone of googletest
Call Stack (most recent call first):
  /usr/share/cmake-3.13/Modules/ExternalProject.cmake:3105 (_ep_add_download_command)
  tests/fetch_and_build_gtest.cmake:5 (externalproject_add)
  tests/CMakeLists.txt:2 (include)

Please allow us to use system googletest.
Thanks

--- a/tests/fetch_and_build_gtest.cmake
+++ b/tests/fetch_and_build_gtest.cmake
@@ -1,6 +1,6 @@
-
+find_package(GTest)
 # Fetch and build GoogleTest if not present in the system
-if(NOT TARGET GTest)
+if(NOT TARGET GTest::GTest)
     include(ExternalProject)
     externalproject_add(
         googletest

